security/pkg/k8s/chiron/utils.go

```diff
@@ -28,12 +28,194 @@ import (
 
 	"github.com/ghodss/yaml"
 
+	"istio.io/istio/pkg/spiffe"
+
+	"istio.io/istio/security/pkg/pki/util"
 	"istio.io/pkg/log"
 
+	cert "k8s.io/api/certificates/v1beta1"
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Generate a certificate and key from k8s CA
+// Working flow:
+// 1. Generate a CSR
+// 2. Submit a CSR
+// 3. Approve a CSR
+// 4. Read the signed certificate
+// 5. Clean up the artifacts (e.g., delete CSR)
+func genKeyCertK8sCA(wc *WebhookController, secretName string, secretNamespace, svcName string) ([]byte, []byte, []byte, error) {
```

Comment:
I highly recommend you split each of the steps into different functions.

Reply:
Done.